langchain4j-core/src/main/java/dev/langchain4j/model/output/OutputParser.java

```diff
@@ -1,21 +1,46 @@
 package dev.langchain4j.model.output;
 
+
+import java.util.Set;
+
 /**
  * Represents an output parser.
  * @param <T> the type of the output.
  */
 public interface OutputParser<T> {
+    /**
+     * Get the supported types.
+     * @return the supported types.
+     */
+    Set<Class<?>> getSupportedTypes();
 
     /**
-     * Parse the given text.
-     * @param text the text to parse.
+     * Parse the given response using a context object.
+     *
+     * @param context the parsing context.
      * @return the parsed output.
      */
-    T parse(String text);
+    T parse(final OutputParsingContext context);
 
     /**
-     * Description of the text format.
-     * @return the description of the text format.
+     * Get the LLM-facing formatting instructions for the output.
+     * @return the formatting instructions.
      */
     String formatInstructions();
```

Comment:
Instead of having formatInstructions and prelude/postlude methods, can we have something like
`void injectParsingInstructions(ChatMessage)`? WDYT?